### PR TITLE
FUSETOOLS-3088 - fix issues when IDs on Rest Elements are changed

### PR DESCRIPTION
**Pull request by bfitzpat**
# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [ ] Did you use the Jira Issue number in the commit comments?
- [ ] Did you set meaningful commit comments on each commit?
- [ ] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [ ] Is there no duplicated code?
- [ ] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [ ] Have you used the correct file header copyright comment?
- [ ] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by bfitzpat** [comment]
This change addresses the ID issues discovered in FUSETOOLS-3087 and FUSETOOLS-3088

**Comment by bfitzpat** [comment]
Good point. Added a couple of new tests @apupier

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java [hunk 1]
```diff
@@ -199,6 +199,90 @@ public void openCamelFileAddRestElementTest() throws Exception {
...
+		);
+		readAndDispatch(20);
+		
+		// ensure that the new rest element (with the changed id) is still selected
+		StructuredSelection secondSsel = (StructuredSelection) restEditor.getSelection();
+		assertThat(secondSsel.getFirstElement()).isInstanceOf(RestElement.class);
+
+		RestElement selection = (RestElement) ssel.getFirstElement();
+		// make sure we got the right one
+		assertThat(selection.getId().equals(newId));
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java]: I think it should be assertThat(selection.getId()).isEqualTo(newId);

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java [hunk 2]
```diff
@@ -199,6 +199,90 @@ public void openCamelFileAddRestElementTest() throws Exception {
...
+		new AddRestElementAction(restEditor, null).run();
+		readAndDispatch(20);
+
+		StructuredSelection ssel = (StructuredSelection) restEditor.getSelection();
+		assertThat(ssel.getFirstElement()).isInstanceOf(RestElement.class);
+
+		RestElement re = (RestElement) ssel.getFirstElement();
+		String newId = "changed-id";
+		re.setId(newId);
+		assertThat(re.getId().equals(newId));
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java]: I think it should be assertThat(re.getId()).isEqualTo(newId);

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java [hunk 3]
```diff
@@ -199,6 +199,90 @@ public void openCamelFileAddRestElementTest() throws Exception {
...
+		assertThat(restVerbElement.getId().equals("myTestID"));
+		restEditor.selectRestVerbElement(restVerbElement);
+		
+		//test to make sure it's removed (there are two REST elements added)
+		new DeleteRestOperationAction(restEditor, null).run();
+		readAndDispatch(20);
+		assertThat(re.getRestOperations().size() == 1);
+		RestVerbElement restVerbElement2 = 
+				(RestVerbElement) re.getRestOperations().values().iterator().next();
+		assertThat(restVerbElement2.getId().equals("myTestID2"));
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java]: I think it should be assertThat(restVerbElement2.getId()).isEqualTo("myTestID2");

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java [hunk 4]
```diff
@@ -199,6 +199,90 @@ public void openCamelFileAddRestElementTest() throws Exception {
...
+		
+		RestVerbElement restVerbElement = 
+				(RestVerbElement) re.getRestOperations().values().iterator().next();
+		assertThat(restVerbElement.getId().equals("myTestID"));
+		restEditor.selectRestVerbElement(restVerbElement);
+		
+		//test to make sure it's removed (there are two REST elements added)
+		new DeleteRestOperationAction(restEditor, null).run();
+		readAndDispatch(20);
+		assertThat(re.getRestOperations().size() == 1);
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java]: i think that it shoudl be assertThat(re.getRestOperations()).hasSize(1);

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java [hunk 5]
```diff
@@ -199,6 +199,90 @@ public void openCamelFileAddRestElementTest() throws Exception {
...
+		readAndDispatch(20);
+
+		Display.getDefault().syncExec(() -> 
+			restEditor.createRestOperation(RestVerbElement.GET_VERB, "test2", "myTestID2", "")
+		);
+		readAndDispatch(20);
+		
+		RestVerbElement restVerbElement = 
+				(RestVerbElement) re.getRestOperations().values().iterator().next();
+		assertThat(restVerbElement.getId().equals("myTestID"));
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java]: assertThat(restVerbElement.getId()).isEqualTo("myTestID");

File: editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java [hunk 6]
```diff
@@ -199,6 +199,90 @@ public void openCamelFileAddRestElementTest() throws Exception {
...
+		
+		// add a REST element
+		new AddRestElementAction(restEditor, null).run();
+		readAndDispatch(20);
+
+		StructuredSelection ssel = (StructuredSelection) restEditor.getSelection();
+		RestElement re = (RestElement) ssel.getFirstElement();
+		String newId = "changed-id";
+		re.setId(newId);
+		assertThat(re.getId().equals(newId));
```
Comment (apupier) [inline editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java]: assertThat(re.getId()).isEqualTo("newId");

**Pull request merged**